Feat: Full stack StatsCounter

adaniel105 opened the pull request:
Employs a script that simply fetches the data from the URL (as provided) of a given Rocket.Chat server, persists as json and displays the json data at runtime.

Closes #59 .

anjy7 commented:
hey @adaniel105, is this ready for review?

adaniel105 commented:
I am a bit unsure on how to handle .env variables idiomatically using svelte, given that we want our app to be serverless. Please review and advise.

anjy7 commented:
@adaniel105 could you please expand on "handle .env variables", I dont completely understand the issue..

Few things that are needed to be implemented in this -
1. Have default values - incase the api call fails or returns null or api key is not provided, we need to have default values. The user should be easily able to hardcode the values.
2. create a `util` folder under src/lib to have these build-time function calls at one place. I dont think src/compiler/scripts/countStats.ts is the right place because it should include only the things that are specific to the compiler.
3. The fetch function doesnt actually run in current implementation. Please do some research and suggest ways in which we can add scripts to the build pipleline such that whenever a user runs `bun build` it first fetches results for the components used - modifies/replaces the "default" values with the latest ones and that is used inside the component.

adaniel105 commented:
@anjy7 I was referring to the fact that the `.env` had to be prefixed with `PUBLIC_` to enable them to be read. But after my changes, i can't even read them in from the .env at all. `src/lib/util/` isn't able to access `$env/static/public`. I will dig through the docs for some older implementations to solve this, but maybe you could provide some guidance. (the env variables used in this pr are just for test) 

- Implemented default values in case the api call fails. Function writes default values to file to be read when failure occurs
- moved script over to util folder
- added `fetchStats` script to pipeline so api can be called and values can be dumped into json, then extracted by `counters.js` :  `"build": "bun src/lib/util/countStats.ts && vite build"`

anjy7 commented:
@adaniel105 
1. we will never use any  env vars in our pages..in situations like this which require data fetching, we will fetch the data during build time and update the json file. This updated json file will be used in the components. Hope this answers the query above, if not, pls lmk.

2. pls dont directly write `bun src/lib/util/countStats.ts` to run the script. We will be having other fetch functions too in future. A script that executes every fetch function file in `src/lib/util` would be better.\

3. I'm not sure about the env vars not working in the new folder path...might be some silly overlook, pls try to debug.

adaniel105 commented:
I was more worried about them being exposed on the repo itself @anjy7 . also will try to debug. Grateful for the feedback as always.

anjy7 commented:
> I was more worried about them being exposed on the repo itself @anjy7 . also will try to debug. Grateful for the feedback as always.

We wont put the env variables in the repo, a sample .env file will be present and the docs will mention the api keys required too.

adaniel105 commented:
@anjy7 ok fixed by using vite directly instead of sveltekit 👍

anjy7 commented:
LGTM @adaniel105 , thank you for the contribution and changes!
